tests/lib.rs

```diff
@@ -85,48 +87,91 @@ impl Storage for MyStorage {
 
 #[test]
 fn new_read() {
-
   let read_size : usize = 4096;
   let mut read_position : usize = 0;
-  let mut index : usize = 0;
+  let mut content_len : usize = 4 * MAX_CHUNK_SIZE as usize;
   let mut my_storage = MyStorage::new();
   let mut data_map = datamap::DataMap::None;
-  let mut original : Vec<u8> = Vec::with_capacity(DATA_SIZE as usize);
-  let mut decrypted : Vec<u8> = Vec::with_capacity(read_size);
-  original = random_bytes(DATA_SIZE as usize);
+  let mut original : Vec<u8> = Vec::with_capacity(content_len);
+  original = random_bytes(content_len);
```

Comment:
But you're returning a `Vec` so this isn't using the allocation from the first line, it's creating a new one and assigning to the slot.